collection/Collections.java

```diff
@@ -101,6 +124,15 @@ public static <A, B, C> Triple<A, B, C> triple(A first, B second, C third) {
         return new Triple<>(first, second, third);
     }
 
+    public static <T> Set<T> intersection(Set<T> set1, Set<T> set2) {
```

Comment:
Let's make it more efficient: avoid creating a copy and just build the set using loops